socketIO_client/heartbeats.py

```diff
@@ -31,7 +31,8 @@ def run(self):
                     interval_in_seconds = self._hurry_interval_in_seconds
                 else:
                     interval_in_seconds = self._relax_interval_in_seconds
-                self._rest.wait(interval_in_seconds)
+                if self._rest and hasattr(self._rest, 'wait'):
```

Comment:
I think there is some weird race condition here when the interpreter is shutting down and threads are running, I sometimes got KeyError here that self doesn't have _rest key. So someone deleted the key before the thread was stopped